x-pack/solutions/security/test/security_solution_api_integration/test_suites/detections_response/rules_management/prebuilt_rules/customization_enabled/customization/detect_customization_with_base_version.ts

```diff
@@ -0,0 +1,635 @@
+        testFieldCustomization({
+          fieldName: 'related_integrations',
+          customizedValue: [{ package: 'package-a', version: '^1.2.3' }],
+        }));
+
+      it('"required_fields" field', () =>
+        testFieldCustomization({
+          fieldName: 'required_fields',
+          customizedValue: [{ name: '@timestamp', type: 'date' }],
+        }));
+
+      it('"max_signals" field', () =>
+        testFieldCustomization({
+          fieldName: 'max_signals',
+          customizedValue: 42,
+        }));
+
+      it('"investigation_fields" field', () =>
+        testFieldCustomization({
+          fieldName: 'investigation_fields',
+          customizedValue: { field_names: ['blob', 'boop'] },
+        }));
+
+      it('"rule_name_override" field', () =>
+        testFieldCustomization({
+          fieldName: 'rule_name_override',
+          customizedValue: 'override string',
+        }));
+
+      it('"timestamp_override" field', () =>
+        testFieldCustomization({
+          fieldName: 'timestamp_override',
+          customizedValue: 'event.ingested',
+        }));
+
+      it('"timeline_template" fields', async () => {
+        const { body: nonCustomizedRule } = await securitySolutionApi
+          .readRule({
+            query: { rule_id: PREBUILT_RULE_ID },
+          })
+          .expect(200);
+
+        const { body: customizedResponse } = await securitySolutionApi
+          .patchRule({
+            body: {
+              rule_id: PREBUILT_RULE_ID,
+              timeline_id: '123',
+              timeline_title: 'timeline title',
+            },
+          })
+          .expect(200);
+
+        expect(customizedResponse.rule_source).toMatchObject({
+          type: 'external',
+          is_customized: true,
+        });
+
+        const { body: customizationRevertedResponse } = await securitySolutionApi
+          .updateRule({
+            body: {
+              ...nonCustomizedRule,
+              id: undefined,
+            },
+          })
+          .expect(200);
+
+        expect(customizationRevertedResponse.rule_source).toMatchObject({
+          type: 'external',
+          is_customized: false,
+        });
+      });
+
+      it('"building_block_type" field', () =>
+        testFieldCustomization({
+          fieldName: 'building_block_type',
+          customizedValue: 'building block string',
+        }));
+
+      describe('rule schedule', () => {
+        it('"interval" field', () =>
+          testFieldCustomization({
+            fieldName: 'interval',
+            customizedValue: '30m',
+          }));
+
+        it('"from" field', () =>
+          testFieldCustomization({
+            fieldName: 'from',
+            customizedValue: 'now-10m',
+          }));
+
+        it('"to" field', () =>
+          testFieldCustomization({
+            fieldName: 'to',
+            customizedValue: 'now-1m',
+          }));
+      });
+    });
+
+    describe('custom query rule fields', () => {
```

Comment:
Probably also want a "saved query rule fields" block in this section, or at least something testing for `saved_id` since the fields are mostly the same between saved query and custom query rule types.